scheduler/src/cook/mesos/pool.clj

```diff
@@ -7,12 +7,26 @@
 (defn check-pool
   "Returns true if requesting-default-pool? and the entity does not have a pool
    or the entity has a pool named pool-name"
-  [source eid entity->pool pool-name requesting-default-pool?]
-  (let [ent (d/entity source eid)
-        pool (entity->pool ent)]
-    (or (and (nil? pool)
-             requesting-default-pool?)
-        (= pool-name (:pool/name pool)))))
+  ([ent entity->pool pool-name requesting-default-pool?]
+   (let [pool (entity->pool ent)]
+     (or (and (nil? pool)
+              requesting-default-pool?)
+         (= pool-name (:pool/name pool)))))
+  ([source eid entity->pool pool-name requesting-default-pool?]
+   (check-pool (d/entity source eid) entity->pool pool-name requesting-default-pool?)))
+
+(def nil-pool (str (UUID/randomUUID)))
+
+(defn check-pool-for-listing
+  "Returns true if either the provided pool-name is the 'nil' pool, or if
+  pool/check-pool returns true. This allows us to return jobs in all pools when the user
+  does not specify a pool."
+  ([entity entity->pool pool-name default-pool?]
+   (or
+     (= nil-pool pool-name)
```

Comment:
Why can't this just be `(nil? pool-name)` and we can avoid the `nil-pool`

Reply:
Because Datomic won't let you have `nil` as a "source" in the datalog query used by `get-active-jobs-by-user-and-state`

Reply:
Ah, gotcha. Annoying.